service/history/workflow/mutable_state_impl.go

```diff
@@ -4936,6 +4937,7 @@ func (ms *MutableStateImpl) GetUpdateCondition() (int64, int64) {
 func (ms *MutableStateImpl) SetSpeculativeWorkflowTaskTimeoutTask(
 	task *tasks.WorkflowTaskTimeoutTask,
 ) error {
+	task.Speculative = true
```

Comment:
Decided to put this here instead of in the 2 callers of this function. Happy to discuss.